raiden/network/transport/matrix/client.py

```diff
@@ -255,6 +255,41 @@ def __init__(
         # can remain constant over multiple loops (if no events occur).
         self.sync_iteration = 0
 
+        self._sync_filter_id: Optional[int] = None
+
+    def create_sync_filter(self, broadcast_rooms: Dict[str, Optional[Room]]) -> None:
+        ignore_rooms = [room.room_id for room in broadcast_rooms.values() if room is not None]
+        filter_params = {
+            "room": {
+                # Filter out all room state events / messages from broadcast rooms
+                "not_rooms": ignore_rooms,
+                # Ignore "message recipts" from all rooms
+                "ephemeral": {"not_types": ["m.receipt"]},
+                # "state": {
```

Comment:
Do I read this as
* all rooms but the ones listed in `not_room`
* all ephemeral events but those of type `m.receipt`
* no state events, because it's not listed in the filter
?

Reply:
Yeah, I'm a bit unclear on that as well. 
I'd be very surprised if the transport worked correctly without at least:
- `m.room.join_rules`
- `m.room.member`
- `m.room.canonical_alias`
- `m.room.aliases`

Reply:
Forget about the "state" rule inside there, i removed it (trying some ideas out).

> Do I read this as
> all rooms but the ones listed in not_rooms
> all ephemeral events but those of type m.receipt

Think about it as after fetching all data:
1. Filter out all updates from `ignore_rooms`.
2. Filter out the remaining updates which have `m.receipt` type

Reply:
So the default is to include everything that is not listed explicitly in the filter? Then why do we need `"presence": {"types": ["m.presence"]}`? Isn't it included by default?

I'm willing to read up on the filter definition if you point me to the right place. I did not find the right documentation when looking myself.

Reply:
Yes, the default is to include everything that is not "excluded" here.

For presence stuff, i guess i just added that as an explicit default. For the documentation, you can see my comment in this PR: https://github.com/raiden-network/raiden/pull/5571#discussion_r363381071